src/Microsoft.DotNet.Build.Tasks/GenerateResourcesCode.cs

```diff
@@ -0,0 +1,239 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Resources;
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateResourcesCode : Task
+    {
+        [Required]
+        public string ResxFilePath { get; set; }
+
+        [Required]
+        public string IntermediateFilePath { get; set; }
+
+        [Required]
+        public string OutputSourceFilePath { get; set; }
+
+        [Required]
+        public string AssemblyName { get; set; }
+
+        public bool DebugOnly { get; set; }
+
+        public override bool Execute()
+        {
+            bool result = true;
+
+            try
+            {
+                _resxReader = new ResXResourceReader(ResxFilePath);
+                _intermediateFile = IntermediateFilePath + ".temp";
+                _targetStream = File.CreateText(_intermediateFile);
+
+                if (String.Equals(Path.GetExtension(OutputSourceFilePath), ".vb", StringComparison.OrdinalIgnoreCase))
+                {
+                    _targetLanguage = TargetLanguage.VB;
+                }
+
+                _keys = new Dictionary<string, int>();
+                WriteClassHeader();
+                RunOnResFile();
+                WriteDebugCode();
+                WriteClassEnd();
+                Close();
+                ProcessTargetFile();
+            }
+            catch (Exception e)
+            {
+                Close();
+
+                Log.LogMessage(e.Message);
+
+                if (e is System.UnauthorizedAccessException)
+                {
+                    Log.LogMessage("The generated {0} file needs to be updated but the file is read-only. Please checkout this file so it can be updated.", OutputSourceFilePath);
```

Comment:
I also get UnauthorizedAccessException when a handle is opened by someone else with a restrictive sharing mode - I'd change the message to say it may be read only.